File: src/test/java/com/shockang/study/algorithm/java/leetcode/util/ArrayUtil.java
package com.shockang.study.algorithm.java.leetcode.util;

import java.util.StringJoiner;

public class ArrayUtil {
	public static int[] stringToIntArray(String s) {
		s = s.trim();
		// [2,7,11,15]
		s = s.substring(1, s.length() - 1);
		String[] splits = s.split(",");
		int n = splits.length;
		int[] res = new int[n];
		for (int i = 0; i < n; i++) {
			res[i] = Integer.parseInt(splits[i]);
		}
		return res;
	}

	public static String intArrayToString(int[] a) {
		StringJoiner sj = new StringJoiner(",", "[", "]");
		for (int i : a) {
			sj.add(String.valueOf(i));
		}
		return sj.toString();
	}
}

File: src/main/scala/com/shockang/study/algorithm/scala/util/ArrayUtil.scala
package com.shockang.study.algorithm.scala.util

import java.util.StringJoiner
import scala.language.implicitConversions

/**
 *
 * @author Shockang
 */
object ArrayUtil {
  implicit def stringToIntArray(str: String): Array[Int] = {
    var s = str.trim()
    // [2,7,11,15]
    s = s.substring(1, s.length - 1)
    val splits = s.split(",")
    val n = splits.length
    val res = new Array[Int](n)
    for (i <- 0 until n) {
      res(i) = splits(i).toInt
    }
    res
  }

  implicit def intArrayToString(a: Array[Int]): String = {
    val sj = new StringJoiner(",", "[", "]")
    for (i <- a) {
      sj.add(String.valueOf(i))
    }
    sj.toString
  }

  def print[T](a: Array[T]): Unit = {
    // scalastyle:off println
    println(a.mkString(","))
    // scalastyle:on println
  }
}
